pkg/apis/serving/v1alpha2/explainer.go

```diff
@@ -67,6 +68,10 @@ func (e *ExplainerSpec) Validate(config *InferenceServicesConfig) error {
 	if err := validateStorageURI(e.GetStorageUri()); err != nil {
```

Comment:
Best to make this extensible.

errs := []error{
  explainer.Validate(config),
  validateStorageURI(e.GetStorageUri()),
  validateResourceRequirements(explainer.GetResourceRequirements),
}
for _, err := range errs {
 if err != nil {
   return err
 }
}
return nil

Reply:
Good suggestion and will refactor. It seems the scope of this PR now is not just to fix the original issue, but to refactor the whole validation logic, I will do this PR with this target now.